### ISSUE TITLE
Invalid VNIC creation parameters for network which does not allow for public IPs

### ISSUE THREAD

**Issue by ehaselwanter** [stephenpearson/kitchen-oci#5]
```shell
kitchen create
-----> Starting Kitchen (v1.20.0)
-----> Creating <default-oracle>...
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Failed to complete #create action: [Invalid VNIC creation parameters] on default-oracle
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

changing the code to

```ruby
      def create_vnic_details(config)
        OCI::Core::Models::CreateVnicDetails.new(
          assign_public_ip: false,
          display_name: 'primary_nic',
          subnetId: config[:subnet_id]
        )
      end
```

works

**Comment by stephenpearson** [comment]
It's possible you are using a subnet which prohibits public IPs.  We could add a simple workaround for the general case and assume that people selecting a private IP do not want a public IP.

We might need an override for that too, but meanwhile could you see if this works for you?

```ruby
      def create_vnic_details(config)
        OCI::Core::Models::CreateVnicDetails.new(
          assign_public_ip: !config[:use_private_ip],
          display_name: 'primary_nic',
          subnetId: config[:subnet_id]
        )
      end
```

**Comment by ehaselwanter** [comment]
yeah. I had that in the title of the issue. it definitely is a network which does not allow for assigning public IPs.

and yes. your solution does work. we then should document the side-effect of this configuration.

```
kitchen converge
-----> Starting Kitchen (v1.20.0)
-----> Creating <default-oracle>...
       Waiting for SSH service on 10.XX.XX.XX:22, retrying in 3 seconds
       [SSH] Established
       Finished creating <default-oracle> (1m9.19s).
-----> Converging <default-oracle>...
       Preparing files for transfer
```

**Comment by stephenpearson** [comment]
I merged #6 which looks up whether the subnet allows VNICs to have a public ip.  If it does, then the VNIC gets a public IP, otherwise it doesn't.  I think this is the least surprising approach for users and should work for your use case seamlessly.

**Comment by ehaselwanter** [comment]
I can confirm that it works.

```shell
kitchen -v
Test Kitchen version 1.20.0

chef gem list |grep oci
kitchen-oci (1.2.0)
oci (2.1.0)
```
